server/tools/img_generators/fal.py

```
@@ -0,0 +1,195 @@
+            mime_type, width, height, extension = await get_image_info_and_save(
+                image_url, 
+                os.path.join(FILES_DIR, f'{image_id}')
+            )
+            
+            filename = f'{image_id}.{extension}'
+            return mime_type, width, height, filename
+            
+        except Exception as e:
+            print(f"Fal AI generation error: {str(e)}")
+            traceback.print_exc()
+            raise Exception(f"Fal AI generation failed: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)